gen/briefinggen.py

```diff
@@ -24,19 +31,144 @@ class CommInfo:
+                f"{self.enemy_base.name}."
+            )
+        elif self.stance in [CombatStance.DEFENSIVE, CombatStance.AMBUSH]:
+            return (
+                "On this location, our ground forces will hold position. We are not expecting an enemy assault."
+            )
+        # TODO: Write a situation description for player RETREAT stance
+        elif self.stance == CombatStance.RETREAT:
+            return ''
+        else:
+            logging.warning('Briefing did not receive a known CombatStance')
+            return ''
+
+    @property
+    def _disadvantage_description(self) -> str:
+        '''Situation description for when player has a numerical disadvantage
+
+        Returns:
+            str
+        '''
+        if self.stance == CombatStance.AGGRESSIVE:
+            return (
+                "On this location, our ground forces will try an audacious "
+                "assault against enemies in superior numbers. The operation"
+                " is risky, and the enemy might counter attack."
+                )
+        elif self.stance == CombatStance.ELIMINATION:
+            return (
+                "On this location, our ground forces will try an audacious assault against "
+                "enemies in superior numbers. The operation is risky, and the enemy might "
+                "counter attack."
+            )
+        elif self.stance == CombatStance.BREAKTHROUGH:
+            return (
+                "On this location, our ground forces have been ordered to rush toward "
+                f"{self.enemy_base.name}. Wish them luck... We are also expecting a counter attack."
+            )
+        elif self.stance in [CombatStance.DEFENSIVE, CombatStance.AMBUSH]:
+            return (
+                "On this location, our ground forces have been ordered to hold still, "
+                "and defend against enemy attacks. An enemy assault might be iminent."
+            )
+        # TODO: Write a situation description for player RETREAT stance
+        elif self.stance == CombatStance.RETREAT:
+            return ''
+        else:
+            logging.warning('Briefing did not receive a known CombatStance')
+            return ''
+
+    @property
+    def brief(self) -> str:
+        '''Situation briefing string
+
+        Returns:
+            str
+        '''
+        if self._zero_units_sentence:
+            return self._zero_units_sentence
+        situation = self._random_frontline_sentence
+        if self.advantage:
+            situation += self._advantage_description
+        else:
+            situation += self._disadvantage_description
+        return situation
+
+
 class MissionInfoGenerator:
     """Base type for generators of mission information for the player.
 
     Examples of subtypes include briefing generators, kneeboard generators, etc.
     """
 
-    def __init__(self, mission: Mission) -> None:
+    def __init__(self, mission: Mission, game: 'Game') -> None:
```

Comment:
with `from __future__ import annotations` you don't need the quotes.

Reply:
I get a circular import exception if I import Game type, even with from __future__ import annotations.  There's probably an underlying issue there, but until it's discovered I'm not sure of another way to type hint it.

Reply:
Whoops, also need:

```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from game import Game
```

at the top. That way it only imports `Game` when it needs it for type checking. It's... kind of gross but it's what we have. (I think when it's just in quotes it's documentation for us but doesn't actually help mypy at all, but not sure)